core/src/com/softwarearchitecture/game_server/states/JoinLobby.java

```diff
@@ -0,0 +1,134 @@
+package com.softwarearchitecture.game_server.states;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.badlogic.gdx.Gdx;
+import com.badlogic.gdx.graphics.g2d.SpriteBatch;
+import com.badlogic.gdx.scenes.scene2d.Stage;
+import com.badlogic.gdx.scenes.scene2d.ui.Label;
+import com.badlogic.gdx.scenes.scene2d.ui.Skin;
+import com.badlogic.gdx.scenes.scene2d.ui.TextField;
+import com.badlogic.gdx.utils.viewport.ScreenViewport;
+import com.softwarearchitecture.GameApp;
+import com.softwarearchitecture.game_server.buttons.Button;
+import com.softwarearchitecture.game_server.buttons.Factory;
+import com.softwarearchitecture.game_server.buttons.Observer;
+import com.softwarearchitecture.game_server.buttons.TypeEnum;
+import com.softwarearchitecture.game_server.buttons.GridLayout;
+import com.softwarearchitecture.math.Rectangle;
+
+public class JoinLobby extends State implements Observer {
+
+    private Stage stage;
+    private TextField lobbyCodeField;
+    private Label enterLobbyLabel;
+    private Skin skin;
+
+    public JoinLobby() {
+        super();
+        List<TypeEnum> buttonTypes = new ArrayList<>();
+        buttonTypes.add(TypeEnum.GAME_MENU);
+        buttons = createButtons(buttonTypes);
+
+        this.stage = new Stage(new ScreenViewport());
+        this.skin = new Skin(Gdx.files.internal("uiskin.json")); // Placeholder
```

Comment:
This file is not in the repository. We need to add it to not crash the application when this state is achieved. 😉